src/web/components/StickyBottomBanner/PuzzlesBanner.tsx

```diff
@@ -0,0 +1,238 @@
+			hasOptedOutOfArticleCount: !props.hasConsentedToArticleCounts,
+		},
+	};
+};
+
+const getPuzzlesBanner = (
+	meta: { [key: string]: any },
+	url: string,
+): Promise<Response> => {
+	const json = JSON.stringify(meta);
+	return fetch(url, {
+		method: 'post',
+		headers: { 'Content-Type': 'application/json' },
+		body: json,
+	});
+};
+
+export const canShow = async ({
+	remotePuzzlesBannerConfig,
+	isSignedIn,
+	asyncCountryCode,
+	contentType,
+	sectionName,
+	shouldHideReaderRevenue,
+	isMinuteArticle,
+	isPaidContent,
+	isSensitive,
+	tags,
+	contributionsServiceUrl,
+	alreadyVisitedCount,
+	engagementBannerLastClosedAt,
+	subscriptionBannerLastClosedAt,
+}: CanShowProps): Promise<CanShowResult> => {
+	const isPuzzlesPage =
+		window.guardian.config.page.section === 'crosswords' ||
+		window.guardian.config.page.series === 'Sudoku';
+
+	if (shouldHideReaderRevenue) {
+		// We never serve Reader Revenue banners in this case
+		return { result: false };
+	}
+
+	if (isPuzzlesPage && remotePuzzlesBannerConfig) {
+		const countryCode = await asyncCountryCode;
+		const hasConsentedToArticleCounts = await getArticleCountConsent();
+		const bannerPayload = buildPayload({
+			isSignedIn,
+			countryCode,
+			contentType,
+			sectionName,
+			shouldHideReaderRevenue,
+			isMinuteArticle,
+			isPaidContent,
+			tags,
+			contributionsServiceUrl,
+			isSensitive,
+			alreadyVisitedCount,
+			engagementBannerLastClosedAt,
+			subscriptionBannerLastClosedAt,
+			hasConsentedToArticleCounts,
+		});
+		return getPuzzlesBanner(
+			bannerPayload,
+			`${contributionsServiceUrl}/puzzles`,
+		)
+			.then(checkForErrors)
+			.then((response) => response.json())
+			.then((json: { data?: any }) => {
+				if (!json.data) {
+					return { result: false };
+				}
+
+				const { module, meta } = json.data;
+
+				return { result: true, meta: { module, meta } };
+			});
+	}
+
+	return { result: false };
+};
+
+type Props = {
+	meta: any;
+	module: { url: string; name: string; props: any[] };
+};
+
+export const PuzzlesBanner = ({ meta, module }: Props) => {
+	const [Banner, setBanner] = useState<React.FC>();
+
+	const [hasBeenSeen, setNode] = useHasBeenSeen({
+		threshold: 0,
+		debounce: true,
+	}) as HasBeenSeen;
+
+	useEffect(() => {
+		if (module === undefined || meta === undefined) {
+			return;
+		}
+
+		window.guardian.automat = {
+			react: React,
+			preact: React,
+			emotionCore,
+			emotionTheming,
+			emotion,
+		};
+
+		window
+			.guardianPolyfilledImport(module.url)
+			.then((bannerModule: { [key: string]: JSX.Element }) => {
+				setBanner(() => bannerModule[module.name]); // useState requires functions to be wrapped
+				sendOphanComponentEvent('INSERT', meta);
+			})
+			.catch((error) =>
+				// eslint-disable-next-line no-console
+				console.log(`banner - error is: ${error}`),
+			);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	// Should only run once
+	useEffect(() => {
+		if (hasBeenSeen && meta) {
```

Comment:
`meta` is set to always be defined
https://github.com/guardian/dotcom-rendering/pull/2736/files#diff-c71689108b52189dab75a54e7f3de25e677987bd0f09d0b30c1a841f136dfb83R164

therefore adding `if(meta)` doesn't make much sense here? unless it isnt always defined? (I am away this is an antipatter used elsewhere)